src/gasEstimator/gasEstimator.ts

```diff
@@ -0,0 +1,50 @@
+import type { BigNumber } from 'ethers';
+import { isDeployRequest } from '../common/relayRequest.utils';
+import type { EnvelopingTxRequest } from '../common/relayTransaction.types';
+import RelayClient from '../RelayClient';
+import {
+  standardMaxPossibleGasEstimation,
+  linearFitMaxPossibleGasEstimation,
+} from './utils';
+
+const estimateRelayMaxPossibleGas = async (
+  request: EnvelopingTxRequest,
+  relayWorkerAddress: string
+): Promise<BigNumber> => {
+  const {
+    relayRequest,
+    metadata: { signature },
+  } = request;
+
+  const { tokenContract, tokenAmount } = relayRequest.request;
+  const { feesReceiver, callForwarder, gasPrice } = relayRequest.relayData;
+
+  const relayClient = new RelayClient();
+
+  const isSmartWalletDeploy = isDeployRequest(relayRequest);
+
+  //FIXME validate how to generate the smart wallet address
+  const preDeploySWAddress = isSmartWalletDeploy ? undefined : undefined;
```

Comment:
To generate the SW Address I could generate my own function or use https://github.com/rsksmart/rif-relay-client/blob/PP601_gasEstimator/src/discovery/utils.ts#L147, but this adds more complexity for a small task.

Reply:
totally agree, we may consider this improvement in future.